tracing.go

```diff
@@ -84,53 +87,155 @@ func (s Span) ParentID() uint64 { return s.parentID }
 // SpanID returns the id of this specific RPC
 func (s Span) SpanID() uint64 { return s.spanID }
 
-// EnableTracing controls whether tracing is enabled for this context
-func (s *Span) EnableTracing(enabled bool) {
-	if enabled {
-		s.flags |= tracingFlagEnabled
-	} else {
-		s.flags &= ^tracingFlagEnabled
+// Flags returns flags bitmap. Interpretation of the bits is up to the tracing system.
+func (s Span) Flags() byte { return s.flags }
+
+// SetTraceID sets traceID
+func (s *Span) SetTraceID(traceID uint64) { s.traceID = traceID }
```

Comment:
I'd like to avoid having these in the public interface. Can we create a custom unexported type, 
```go
type injectableSpan Span
```

And have those functions defined on that `injectableSpan`. We can freely cast between them since they're the same underlying type, but the public interface should not have those functions exposed anywhere